python/cudf/cudf/core/dataframe.py

```diff
@@ -6334,10 +6351,16 @@ def _apply_support_method(self, method, axis=0, *args, **kwargs):
                 "support `bool_only`."
                 raise NotImplementedError(msg)
 
-            prepared = self._prepare_for_rowwise_op()
+            prepared = self._prepare_for_rowwise_op(method)
+            prepared = prepared.astype("float64")
+            prepared = prepared.fillna(np.nan)
```

Comment:
`null` and `NaN` are two distinctly different values and don't necessarily have the same behavior where this isn't really safe.

Reply:
This `fillna` with `nan` was used because in `cupy` we only have `nan`'s and the `skipna` parameter of all the ops is equivalent to what `nanmin`, `nanmax`, ..etc ops perform. I think keeping `skipna` param in view this specific case it would make sense to fill null values with `nan` values. Also for a column filled with nulls we seem to return `nan` as scalars:

```python
>>> s = cudf.Series([None, None, None], dtype='int64')
>>> s
0    <NA>
1    <NA>
2    <NA>
dtype: int64
>>> s.min()
nan
>>> s.max()
nan
>>> s.var()
nan
```

Reply:
There's plans to change this in the future to return a `cudf.Scalar` object as well.

Reply:
I see, so should we switch the dataframe implementations to transpose(when axis=1) and call the `Series` ops implementations?

Reply:
> I see, so should we switch the dataframe implementations to transpose(when axis=1) and call the `Series` ops implementations instead of going through cupy?

That would be very slow to the point of being a non-starter. In hindsight, I think the ops you have defined above should be okay for null behavior as well, but we should still differentiate between `NaN` and `null` if possible and cast back to the original dtype.

Reply:
@kkraus14 `null` and `nan` are now being handled by extracting out null mask per column and applying it back to the cupy result.